File: framework/apis/api_authenticated_token_chaining_api.py
from playwright.sync_api import APIResponse

from framework.core.base_api import BaseAPI


class ApiAuthenticatedTokenChainingApi:
    def __init__(self, client: BaseAPI) -> None:
        self.client = client

    def login(self, body: dict) -> APIResponse:
        return self.client.post("/auth/login", body)

    def current_user(self) -> APIResponse:
        response = self.client.get("/auth/me")
        self.client.assert_status(response, 200)
        return response

File: packages/test-framework/apis/ApiAuthenticatedTokenChainingApi.ts
import { APIResponse } from '@playwright/test';
import { BaseAPI } from '@core/BaseAPI';

/**
 * Generated API client — API Authenticated Token Chaining
 * @generated WebPilot
 */
export class ApiAuthenticatedTokenChainingApi {
  constructor(private readonly client: BaseAPI) {}

  async postApibaseurlauthlogin(body: unknown): Promise<APIResponse> {
    const response = await this.client.post('/auth/login', body);
    return response;
  }

  async getApibaseurlauthme(): Promise<APIResponse> {
    const response = await this.client.get('/auth/me');
    await this.client.assertStatus(response, 200);
    return response;
  }
}
